lib/spack/spack/schema/modules.py

```diff
@@ -135,7 +135,7 @@
                 'default': [],
                 'items': {
                     'type': 'string',
-                    'enum': ['tcl', 'dotkit', 'lmod']}},
+                    'enum': ['tcl', 'lmod']}},
```

Comment:
Same issue here -- removing this from the schema is going to break potentially valid configurations where dotkit was being generated before.  Just warn.

Reply:
See above.